### PR TITLE
CAKE-5330 | Affiliate Integration Branch

### PR DESCRIPTION
**Pull request by JoshuaRogan**
@Wikia/cake

### CODE REVIEW

File: extensions/wikia/AffiliateService/js/ext.AffiliateService.js [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+], function ($, w, geo, log, units) {
+	'use strict';
+
+	var deferred = $.Deferred();
+
+	var AffiliateService = {
+		$infoBox: undefined,
+
+		canDisplayUnit: function () {
+			typeof AffiliateService.$infoBox !== 'undefined';
```
Comment (cschucode) [inline extensions/wikia/AffiliateService/js/ext.AffiliateService.js]: Does this need a `return` statement?

File: extensions/wikia/AffiliateService/js/ext.AffiliateService.js [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+			// only select paragraphs one level from the root main element
+			var $paragraphs = $('#mw-content-text > p');
+
+			// prepend the unit after the first paragraph below the infobox
+			$paragraphs.each(function (index, element) {
+				var $paragraph = $(element);
+				var paragraphHeight = $paragraph.offset().top;
+
+				if (paragraphHeight > startHeight) {
+					$paragraph.prepend('<div style="background: red; width: 100%; height: 100px"> </div>')
```
Comment (cschucode) [inline extensions/wikia/AffiliateService/js/ext.AffiliateService.js]: This line could use a semi-colon.

File: extensions/wikia/AffiliateService/js/ext.AffiliateService.js [hunk 3]
```diff
@@ -0,0 +1,377 @@
...
+					return true;
+				}
+
+				// do we have debug targeting
+				if (AffiliateService.getDebugTargeting() !== false) {
+					return true;
+				}
+			}
+
+			// tough luck
```
Comment (cschucode) [inline extensions/wikia/AffiliateService/js/ext.AffiliateService.js:119]: Humor?

File: extensions/wikia/AffiliateService/js/ext.AffiliateService.js [hunk 4]
```diff
@@ -0,0 +1,377 @@
...
+			function isValidSlot(yStart) {
+				// if there are no images always return true
+				if (notAllowedYStart.length === 0) {
+					return true;
+				}
+
+				// find the index of where this would be inserted
+				var index = insertSpot(notAllowedYStart, yStart);
+
+				// no images yet, we are gucci
```
Comment (cschucode) [inline extensions/wikia/AffiliateService/js/ext.AffiliateService.js:299]: Making gucci an industry standard?

File: extensions/wikia/AffiliateService/js/ext.AffiliateService.js [hunk 5]
```diff
@@ -0,0 +1,377 @@
...
+					log('Affiliate Unit inserted using fallback slot');
+					$insertionPoint = $fallbackParagraph;
+					return false;
+				}
+			});
+
+			AffiliateService.insertAtPointAndTrack($insertionPoint, unit);
+		},
+
+		// Using mustache to render template and unit info
```
Comment (cschucode) [inline extensions/wikia/AffiliateService/js/ext.AffiliateService.js:356]: Not using mustache anymore.

**Review by vforge — APPROVED** [review]

**Pull request merged**